pkg/commands/docker.go

```diff
@@ -91,7 +93,13 @@ func (t *dockerBuildTask) ScanForDependencies(runner build.Runner) ([]build.Imag
 }
 
 func (t *dockerBuildTask) Build(runner build.Runner) error {
-	args := []string{"build"}
+	// --pull to make sure the local base image cache is up to date.
+	args := []string{"build --pull"}
```

Comment:
make this a parameter too, just default it to true

Reply:
Thought about it. I deliberately hide it to reduce the parameter numbers since we don't need it and requires higher-up changes which is simply defaulted to true. The only case I can think of is users manually copy the base image and they don't have access the registry. I can certainly make the change if you think it may be the case we want to target in the future.

Reply:
that would make sense if we were writing code just for our code base, but this is open source, it should be a parameter for other people to use it if they want to, just how Docker doesn't make an assumption about it